src/psyclone/psyir/nodes/reference.py

```diff
@@ -191,6 +193,57 @@ def datatype(self):
             return UnresolvedType()
         return self.symbol.datatype
 
+    def previous_access(self):
+        '''
+        :returns: the previous reference to the symbol accessed by this
+                  Reference.
+        :rtype: :py:class:`psyclone.psyir.nodes.Node` or None
+        '''
+        # Avoid circular import
+        # pylint: disable=import-outside-toplevel
+        from psyclone.psyir.nodes.routine import Routine
+        # The scope is as far as the Routine that contains this
+        # Reference.
+        var_access = VariablesAccessInfo(nodes=self.ancestor(Routine))
```

Comment:
I would like this be generic to any PSyIR subtree, even when not connected to a Routine. So, I agree that in general we should look for the ancestor(Routine), but when this is not found (e.g we have a standalone statement `a = b + a + b` ) we could start from the expression root.

Also add a test with a disconnected statement, as currently they are all full source trees.

Reply:
Added the code to do it, will work on a test as well.